src/MagnumPlugins/BasisImporter/BasisImporter.cpp

```diff
@@ -0,0 +1,251 @@
+    FROM, OUT OF OR IN CONNETCION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "BasisImporter.h"
+
+#include <basisu_transcoder.h>
+
+#include <Corrade/Containers/Optional.h>
+#include <Corrade/Containers/StaticArray.h>
+#include <Corrade/Utility/ConfigurationGroup.h>
+#include <Corrade/Utility/ConfigurationValue.h>
+#include <Corrade/Utility/Debug.h>
+#include <Corrade/Utility/String.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Trade/ImageData.h>
+
+namespace Corrade { namespace Utility {
+
+/* Configuration value implementation for BasisTranscodingType */
+template<> struct ConfigurationValue<Magnum::Trade::BasisTranscodingType> {
+
+    static std::string toString(
+        const Magnum::Trade::BasisTranscodingType& value,
+        ConfigurationValueFlags)
+    {
+        #define _c(value) case Magnum::Trade::BasisTranscodingType::value: return #value;
+        switch(value) {
+            _c(Etc1)
+            _c(Etc2)
+            _c(Bc1)
+            _c(Bc3)
+            _c(Bc4)
+            _c(Bc5)
+            _c(Bc7M6OpaqueOnly)
+            _c(Pvrtc1_4OpaqueOnly)
+        }
+        #undef _c
+
+        return "<invalid>";
+    }
+
+    static Magnum::Trade::BasisTranscodingType fromString(
+        const std::string& value, ConfigurationValueFlags)
+    {
+        const Containers::StaticArray<8, const char*> FormatNames{
+            "Etc1", "Etc2", "Bc1", "Bc3", "Bc4", "Bc5",
+            "Bc7M6OpaqueOnly", "Pvrtc1_4OpaqueOnly"
+        };
+
+        Magnum::Int i = 0;
+        for(const char* name: FormatNames) {
+            if(value == name) return Magnum::Trade::BasisTranscodingType(i);
+            ++i;
+        }
+
+        return Magnum::Trade::BasisTranscodingType(-1);
+    }
+};
+
+}}
+
+namespace Magnum { namespace Trade {
+
+namespace {
+
+/* Map BasisTranscodingType to Magnum::CompressedPixelFormat */
+CompressedPixelFormat textureFormat(BasisTranscodingType type, bool hasAlpha) {
+    switch(type) {
+        case BasisTranscodingType::Etc1:
+        case BasisTranscodingType::Etc2:
+            return (hasAlpha) ? CompressedPixelFormat::Etc2RGBA8Unorm
+                : CompressedPixelFormat::Etc2RGB8Unorm;
+        case BasisTranscodingType::Bc1:
+            return (hasAlpha) ? CompressedPixelFormat::Bc1RGBAUnorm
+                : CompressedPixelFormat::Bc1RGBUnorm;
+        case BasisTranscodingType::Bc3:
+            return CompressedPixelFormat::Bc3RGBAUnorm;
+        case BasisTranscodingType::Bc4:
+            return CompressedPixelFormat::Bc4RUnorm;
+        case BasisTranscodingType::Bc5:
+            return CompressedPixelFormat::Bc5RGUnorm;
+        case BasisTranscodingType::Bc7M6OpaqueOnly:
+            return CompressedPixelFormat::Bc7RGBAUnorm;
+        case BasisTranscodingType::Pvrtc1_4OpaqueOnly:
+            return CompressedPixelFormat::PvrtcRGB4bppUnorm;
+        default:
+            CORRADE_ASSERT_UNREACHABLE();
```

Comment:
`CORRADE_ASSERT_UNREACHABLE(); /* LCOV_EXCL_LINE */` will make it squeeze the last bits of code coverage :)